### PR TITLE
Add Ui Class Diagram to DG, more JUnit

### PR DESCRIPTION
**Pull request by shawntangy**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -59,6 +60,22 @@ The rest of the App consists of four components.
 * [**`Storage`**](): Reads data from, and writes data to, the hard disk.
 
 ## Implementations
+
+### Ui Class
+
+The Ui class is represented by the image below. The Ui class handles user input and
+displaying of relevant information including error messages for the program. In the diagram below,
```
Comment (pyongjie) [inline docs/DeveloperGuide.md]: Perhaps "the displaying of relevant information...." would be a better phrasing?

File: docs/images/UiClassDiagram.png
Comment (pyongjie) [inline docs/images/UiClassDiagram.png:1]: Based on our code, I think the main method should be "+Main()" in the diagram?

**Review by pyongjie — COMMENTED** [review]
Codes added for JUnit tests seems generally correct. Updates to DG are more or less alright except for a few minor issues. After the issues are resolved, you can proceed to merge this pull request into our master branch.

**Comment by shawntangy** [comment]
> Codes added for JUnit tests seems generally correct. Updates to DG are more or less alright except for a few minor issues. After the issues are resolved, you can proceed to merge this pull request into our master branch.

Thanks for the review!

**Pull request merged**